### DISCUSSION TITLE
SelectOneMenu are flickering the label briefly

### DISCUSSION THREAD

**Discussion by Zearys** [primefaces/community#2502] [PrimeFaces] [answered]
Hi, 

Been a while that I notice something that always trigger my eyes, but never took the time to post it.
The issue is in PF12, 13 and 14(maybe 11 too, but can't remember). The SelectOneMenu are flickering the label briefly then switch to first item when you enter the page or refresh it.

The issue is present in the showcase of the component:
[https://www.primefaces.org/showcase/ui/input/oneMenu.xhtml](https://www.primefaces.org/showcase/ui/input/oneMenu.xhtml)

To replicate: 
Simply stare the content of a list and press "F5". In the showcase, you'll see rapidly the content of the first list go from "Basic" to "Select One". All the lists in the page are going to display briefly their label then their first value. Repeat if you miss the flickering. 

When there's a lot of SelectOneMenu in a page, like the demo, it's more noticable. My apps tend to have +10 SelectOneMenu in search options and the flickering bother me and the users :).

Is it the aria-label? Does someone have a fix to avoid this issue?

**Comment by melloware** [comment, marked as answer]
Yeah it probably renders with the default and then some Javascript sets it on load of the component that would explain the flicker.

**Comment by Zearys** [comment]
That make sense thanks for replying. Hope it get fix eventually 😊

**Reply by melloware** [reply]
@Zearys it is quite complex Javascript to handle all the scenarios so not sure there "is" a fix.

```js
/**
     * Sets the label text that indicates the currently selected item to the item with the given value.
     * @private
     * @param {string} value Value of the item that was selected.
     */
    setLabel: function(value) {
        var displayedLabel = this.getLabelToDisplay(value);
        var valueEmpty = value === '&nbsp;' || value === '\xa0';
        if (this.cfg.editable) {
            if (valueEmpty)
                this.label.val('');
            else
                this.label.val(displayedLabel);

            var hasPlaceholder = this.label[0].hasAttribute('placeholder');
            this.updatePlaceholderClass((hasPlaceholder && valueEmpty));
        }
        else if (this.cfg.alwaysDisplayLabel && this.cfg.label) {
            this.label.text(this.cfg.label);
        }
        else {
            var labelText = this.label.data('placeholder');
            if (labelText == null || labelText == "") {
                labelText = '&nbsp;';
            }

            this.updatePlaceholderClass((valueEmpty && labelText !== '&nbsp;'));

            if (valueEmpty) {
                if (labelText != '&nbsp;') {
                    this.label.text(labelText);
                } else {
                    this.label.html(labelText);
                }
            }
            else {
                this.label.removeClass('ui-state-disabled');

                var option = null;
                if (this.items) {
                    var selectedItem = this.items.filter('[data-label="' + $.escapeSelector(value) + '"]');
                    option = this.options.eq(this.resolveItemIndex(selectedItem));
                }
                else {
                    option = this.options.filter(':selected');
                }

                if (option && option.data('escape') === false) {
                    this.label.html(displayedLabel);
                } else {
                    this.label.text(displayedLabel);
                }
                this.label.attr('aria-label', displayedLabel);
            }
        }
    },
```
